### ISSUE TITLE
Vagrant up not functional

### ISSUE THREAD

**Issue by syffs** [actionml/chef-pio#1]
I'm facing the following issue right after cloning:
```bash
[user@server chef-pio]$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
/opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- bundler (LoadError)
        from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/util/env.rb:1:in `<top (required)>'
        from /home/user/.vagrant.d/gems/2.2.5/gems/vagrant-berkshelf-5.1.1/lib/vagrant-berkshelf/helpers.rb:62:in `berks'
        from /home/user/.vagrant.d/gems/2.2.5/gems/vagrant-berkshelf-5.1.1/lib/vagrant-berkshelf/action/check.rb:33:in `berkshelf_version_check!'
        from /home/user/.vagrant.d/gems/2.2.5/gems/vagrant-berkshelf-5.1.1/lib/vagrant-berkshelf/action/check.rb:16:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builder.rb:116:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:in `block in run'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/util/busy.rb:19:in `busy'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:in `run'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builtin/call.rb:53:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builder.rb:116:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:in `block in run'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/util/busy.rb:19:in `busy'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:in `run'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:225:in `action_raw'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:200:in `block in action'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/environment.rb:567:in `lock'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:186:in `call'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:186:in `action'
        from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run
```

Any idea?

**Comment by pferrel** [comment]
The chef script is written to deploy to Ubuntu and Amazon Linux, not virtualbox + linux. You may have to modify it to work there.

On Dec 19, 2016, at 3:18 AM, Syffs <notifications@github.com> wrote:

I'm facing the following issue right after cloning:
[user@server chef-pio]$ vagrant up Bringing machine 'default' up with 'virtualbox' provider... /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:inrequire': cannot load such file -- bundler (LoadError)
from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in require' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/util/env.rb:1:in<top (required)>'
from /home/jbessaguet/.vagrant.d/gems/2.2.5/gems/vagrant-berkshelf-5.1.1/lib/vagrant-berkshelf/helpers.rb:62:in berks' from /home/jbessaguet/.vagrant.d/gems/2.2.5/gems/vagrant-berkshelf-5.1.1/lib/vagrant-berkshelf/action/check.rb:33:inberkshelf_version_check!'
from /home/jbessaguet/.vagrant.d/gems/2.2.5/gems/vagrant-berkshelf-5.1.1/lib/vagrant-berkshelf/action/check.rb:16:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builtin/config_validate.rb:25:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:95:in block in finalize_action' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builtin/handle_box.rb:56:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:95:inblock in finalize_action'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builder.rb:116:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:inblock in run'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/util/busy.rb:19:in busy' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:inrun'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builtin/call.rb:53:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/warden.rb:34:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/builder.rb:116:in call' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:inblock in run'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/util/busy.rb:19:in busy' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/action/runner.rb:66:inrun'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:225:in action_raw' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:200:inblock in action'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/environment.rb:567:in lock' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:186:incall'
from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/machine.rb:186:in action' from /opt/vagrant/embedded/gems/gems/vagrant-1.9.0/lib/vagrant/batch_action.rb:82:inblock (2 levels) in run'
`
Any idea?

**Comment by dennybaa** [comment]
Chef cookbook can deploy PIO to Ubuntu using vagrant, however your 
setup of vagrant machine is flawed I guess. The thing is that you 
should have bundler gem installed but your vagrant installation doesn't 
have it...

try
/opt/vagrant/embedded/bin/gem list | grep bundler... to figure out if 
it's there.

But actually you should refer to vagrant installation manual.

- Denis

**Comment by dennybaa** [comment]
@Syffs Moreover if you were using the *Vagrantfile* from this repository, I need to note that it was auto generated. We are not using this Vagrantfile it should be reworked, PRs are welcome :)

**Comment by syffs** [comment]
Thanks for your feedbacks!

> try
/opt/vagrant/embedded/bin/gem list | grep bundler... to figure out if
it's there.

You are right, I've figured this out since

I only had to add a few lines to the Vagrantfile to make a big jump. It now fails while building mahout. I've built a quick Dockerfile and it struggles one the exact same error:

```
==> default: [WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
==> default: Detected Maven Version: 3.0.5 is not in the allowed range [3.3.3,).
==> default: [INFO] ------------------------------------------------------------------------
==> default: [INFO] Reactor Summary:
==> default: [INFO]
==> default: [INFO] Mahout Build Tools ................................ FAILURE [4.850s]
==> default: [INFO] Apache Mahout ..................................... SKIPPED
==> default: [INFO] Mahout Math ....................................... SKIPPED
==> default: [INFO] Mahout HDFS ....................................... SKIPPED
==> default: [INFO] Mahout Map-Reduce ................................. SKIPPED
==> default: [INFO] Mahout Integration ................................ SKIPPED
==> default: [INFO] Mahout Examples ................................... SKIPPED
==> default: [INFO] Mahout Math Scala bindings ........................ SKIPPED
==> default: [INFO] Mahout Spark bindings ............................. SKIPPED
==> default: [INFO] Mahout Flink bindings ............................. SKIPPED
==> default: [INFO] Mahout Spark bindings shell ....................... SKIPPED
==> default: [INFO] Mahout Release Package ............................ SKIPPED
==> default: [INFO] Mahout H2O backend ................................ SKIPPED
==> default: [INFO] ------------------------------------------------------------------------
==> default: [INFO] BUILD FAILURE
==> default: [INFO] ------------------------------------------------------------------------
==> default: [INFO] Total time: 7.714s
==> default: [INFO] Finished at: Tue Dec 20 10:40:01 UTC 2016
==> default: [INFO] Final Memory: 12M/29M
==> default: [INFO] ------------------------------------------------------------------------
==> default: [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.4:enforce (enforce-versions) on project mahout-buildtools: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
==> default: [ERROR]
==> default: [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
==> default: [ERROR] Re-run Maven using the -X switch to enable full debug logging.
==> default: [ERROR]
==> default: [ERROR] For more information about the errors and possible solutions, please read the following articles:
==> default: [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
==> default: STDERR:
==> default: ---- End output of mvn clean install -DskipTests ----
==> default: Ran mvn clean install -DskipTests returned 1
==> default: [2016-12-20T10:40:01+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
```

> @Syffs Moreover if you were using the Vagrantfile from this repository, I need to note that it was auto generated. We are not using this Vagrantfile it should be reworked, PRs are welcome :)

I'll see if I get anywhere in my sandbox, but if I do of course I'll share!

**Comment by dennybaa** [comment]
@Syffs are you using `pio::aio` recipe? We haven't seen such an issue. Maybe upstream failures...

**Comment by syffs** [comment]
> @Syffs are you using pio::aio recipe? We haven't seen such an issue. Maybe upstream failures...

I was using pio::default (the one setup by default in the Vagrantfile), but trying to use pio::aio instead, it seems that using openjdk as in this line:
```
node.override["java"]["install_flavor"] = 'openjdk'
```
causes the below issue:
```
==> default: :::: ERRORS
==> default:
==> default:    Server access Error: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors                                       parameter must be non-empty url=https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/0.13.7/ivys/ivy.xml
==> default:
==> default:                    Server access Error: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException:                                       the trustAnchors parameter must be non-empty url=https://repo1.maven.org/maven2/org/scala-sbt/sbt/0.13.7/sbt-0.13.7.pom
==> default:
==> default:                    Server access Error: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException:                                       the trustAnchors parameter must be non-empty url=https://repo1.maven.org/maven2/org/scala-sbt/sbt/0.13.7/sbt-0.13.7.jar
==> default:
==> default:
==> default:               :: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
==> default:               unresolved dependency: org.scala-sbt#sbt;0.13.7: not found
==> default:               Error during sbt execution: Error retrieving required libraries
==> default:                 (see /home/aml/.sbt/boot/update.log for complete log)
==> default:               Error: Could not retrieve sbt 0.13.7
==> default:
==> default:
==> default:
==> default: ================================================================================
==> default:     Error executing action `run` on resource 'execute[make-distribution.sh]'
==> default:     ================================================================================
==> default:
==> default:     Mixlib::ShellOut::ShellCommandFailed
==> default:     ------------------------------------
==> default:     Expected process to exit with [0], but received '1'
```
Now, commenting the openjdk line, leads to the same mahout errror mentioned before. You should be able to reproduce by using the [Vagrantfile](https://github.com/actionml/chef-pio/files/664774/Vagrantfile.txt) attached

**Comment by syffs** [comment]
FYI changing ubuntu version to a more recent one like 16.04 instead of 14.04 makes it work! There must be some issue with this particular 3.0.X versions of maven available through 14.04 repos. Another option would be to install maven differently to get a newer version.

No other issue through vagrant, rest of the build is successful even with the openjdk line below
```
node.override["java"]["install_flavor"] = 'openjdk'
```
I'll give a try to make it all work with Docker as well

**Comment by dennybaa** [comment]
@Syffs Thanks for the information, indeed we haven't done much testing on anything except 16.04.

**Comment by dennybaa** [comment]
@pferrel @alexice There's indeed dependency issue when running `make-distribution.sh` on 14.04:
```
==> default: STDOUT: Building binary distribution for PredictionIO 0.10.0-incubating...
==> default: Attempting to fetch sbt
==> default: Launching sbt from /opt/data/pio/sbt/sbt-launch-0.13.7.jar
==> default: Getting org.scala-sbt sbt 0.13.7 ...
==> default: 
==> default: :: problems summary ::
==> default: :::: WARNINGS
==> default: 		module not found: org.scala-sbt#sbt;0.13.7
==> default: 
==> default: 	==== local: tried
==> default: 
==> default: 	  /home/aml/.ivy2/local/org.scala-sbt/sbt/0.13.7/ivys/ivy.xml
==> default: 
==> default: 	  -- artifact org.scala-sbt#sbt;0.13.7!sbt.jar:
==> default: 
==> default: 	  /home/aml/.ivy2/local/org.scala-sbt/sbt/0.13.7/jars/sbt.jar
==> default: 
==> default: 	==== typesafe-ivy-releases: tried
==> default: 
==> default: 	  https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/0.13.7/ivys/ivy.xml
==> default: 
==> default: 	==== Maven Central: tried
==> default: 
==> default: 	  https://repo1.maven.org/maven2/org/scala-sbt/sbt/0.13.7/sbt-0.13.7.pom
==> default: 
==> default: 	  -- artifact org.scala-sbt#sbt;0.13.7!sbt.jar:
==> default: 
==> default: 	  https://repo1.maven.org/maven2/org/scala-sbt/sbt/0.13.7/sbt-0.13.7.jar
==> default: 
==> default: 		::::::::::::::::::::::::::::::::::::::::::::::
==> default: 
==> default: 		::          UNRESOLVED DEPENDENCIES         ::
==> default: 
==> default: 		::::::::::::::::::::::::::::::::::::::::::::::
==> default: 
==> default: 		:: org.scala-sbt#sbt;0.13.7: not found
==> default: 
==> default: 		::::::::::::::::::::::::::::::::::::::::::::::
==> default: 
==> default: 
==> default: :::: ERRORS
```

You might have a look into it.

@Syffs Usage of maven *3.3.9-001* on trusty, doesn't solve the above issue ^^^.

**Comment by jbessaguet** [comment]
@dennybaa You're right. Now commenting the below line (usage of openjdk) did solve the issue for me though
```
node.override["java"]["install_flavor"] = 'openjdk'
```

**Comment by syffs** [comment]
@dennybaa in my test runs, that's disabling the usage of openjdk which solved this issue
```
node.override["java"]["install_flavor"] = 'openjdk'
```
However I didn't dig further to understand the root cause

**Comment by dennybaa** [comment]
@Syffs do you have any updates? I'm going to close this. We just do not support 14.04 at the moment, only 16.04.

**Comment by jsyrjala** [comment]
Dependency issues with make-distribution.sh in Ubuntu 14.04 seem to be related OpenJDK certificate authority configuration, see #3.

**Comment by dennybaa** [comment]
resolved in 98c488f90399cfeb1e94aaea2ddc3936e6ec3653
